File: twoSum.py
"""
    Given an array of integers, return indices of the two numbers such that they add up to a specific target.
    You may assume that each input would have exactly one solution, and you may not use the same element twice.

    Given nums = [2, 7, 11, 15], target = 9,
    Because nums[0] + nums[1] = 2 + 7 = 9,
    return [0, 1].

    2018-7-26 19:34:10
"""


class twoSum:
    def findSumIndices(self, nums, target):
        result = []
        for i in range(len(nums)):
            oneNum = nums[i]
            for j in range(i + 1, len(nums)):
                twoNum = nums[j]
                if oneNum + twoNum == target:
                    result.append(i)
                    result.append(j)

        return result


calculator = twoSum()
result = calculator.findSumIndices([2, 7, 0,9, 15], 9)
print(result)

File: leecode-cn/Offer60-TwoSum/TwoSum.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution
{
public:
	vector<double> twoSum(int n);
};

int main()
{
	int n = 2;
	Solution s;
	vector<double> probabilities = s.twoSum(n);
	cout << probabilities.size();

	return 0;
}

vector<double> Solution::twoSum(int n)
{
	vector<vector<double>> dp(n + 1, vector<double>((n + 1) * 6));

	for (int i = 1; i <= n + 1; ++i)
	{
		for (int j = 1 * i; j <= 6 * i; ++j)
		{
			dp[i][j] = 



		}


	}

	return vector<double>();
}
